### PR TITLE
add LLVMExtra protoype

### PR DESCRIPTION
**Pull request by vchuravy**
cc: @wsmoses @maleadt

For now this is hostest at https://github.com/vchuravy/LLVMExtra and only contains the extras from Julia,
I will add my recent PRs for extended OrcV2 later.

### CODE REVIEW

**Comment by maleadt** [comment]
Great stuff! Should we make it part of the LLVM.jl repo instead, because there will be quite some coupling between this API and the wrappers?

File: L/LLVMExtra/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+script = raw"""
+cd LLVMExtra
+CMAKE_FLAGS=()
+# Release build for best performance
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=RelWithDebInfo)
+# Install things into $prefix
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+# Explicitly use our cmake toolchain file and tell CMake we're cross-compiling
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING:BOOL=ON)
```
Comment (giordano) [inline L/LLVMExtra/build_tarballs.jl:24]: Is this really needed?  We should already be setting this variable correctly: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/5684f204cc503b746a032536b90a0667745589bd/src/BuildToolchains.jl#L36-L45

File: L/LLVMExtra/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# Build the library
+CMAKE_FLAGS+=(-DBUILD_SHARED_LIBS=ON)
+cmake -B build -S . -GNinja ${CMAKE_FLAGS[@]}
+ninja -C build -j ${nproc} install
+"""
+
+function configure(julia_version, llvm_version)
+    # These are the platforms we will build for by default, unless further
+    # platforms are passed in on the command line
+    platforms = expand_cxxstring_abis(supported_platforms())
```
Comment (giordano) [inline L/LLVMExtra/build_tarballs.jl:38]: Would this work?
```suggestion
    platforms = expand_cxxstring_abis(supported_platforms(; experimental=julia_version>=v"1.7"))
```

Comment (vchuravy) [reply L/LLVMExtra/build_tarballs.jl:38]: 1.6 should also work no?

File: L/LLVMExtra/build_tarballs.jl (hunk 1 above)
Comment (vchuravy) [reply L/LLVMExtra/build_tarballs.jl:24]: I stole what we have for Enzyme, so it might also work

File: L/LLVMExtra/build_tarballs.jl (hunk 2 above)
Comment (giordano) [reply L/LLVMExtra/build_tarballs.jl:38]: Uh, I thought we didn't have LLVM 11 for those platforms, but we do, yes.
```suggestion
    platforms = expand_cxxstring_abis(supported_platforms(; experimental=true))
```

**Comment by vchuravy** [comment]
> Great stuff! Should we make it part of the LLVM.jl repo instead, because there will be quite some coupling between this API and the wrappers?

Yeah I was hosting it separately for now do that I don't feel to bad about messing about with Git until I get this working.

**Pull request merged**